src/planner/expand_hypertable.c

```diff
@@ -327,10 +337,53 @@ constify_timestamptz_op_interval(PlannerInfo *root, OpExpr *constraint)
 									constraint->inputcollid);
 }
 
+static Datum
+calculate_timestamptz_timezone_bucket(int strategy, Datum width, Datum value, List *args)
+{
+	Assert(list_length(args) == 5);
+
+	if (TIMESTAMP_NOT_FINITE(DatumGetTimestampTz(value)))
+		return value;
+
+	Const *tzname = castNode(Const, lthird(args));
+	Const *offset = castNode(Const, lfifth(args));
+	Datum new_timestamp;
+	bool have_offset = (!offset->constisnull);
+
+	new_timestamp = DirectFunctionCall2(timestamptz_zone, tzname->constvalue, value);
+
+	if (have_offset)
+	{
+		/* Apply offset. */
+		new_timestamp =
+			DirectFunctionCall2(timestamp_mi_interval, new_timestamp, offset->constvalue);
+	}
+
+	/*
+	 * If the interval has a day component then the calculation needs
+	 * to take into account daylight saving time switches and thereby a
+	 * day would not always be exactly 24 hours. We mitigate this by
+	 * adding a safety buffer to account for these dst switches when
+	 * dealing with intervals with day component. These calculations
+	 * will be repeated with exact values during execution.
+	 * Since dst switches seem to range between -1 and 2 hours we set
+	 * the safety buffer to 4 hours.
+	 */
+	TimestampTz new_value = DatumGetTimestampTz(new_timestamp);
+
+	if (strategy == BTLessStrategyNumber || strategy == BTLessEqualStrategyNumber)
+		new_value += 4 * USECS_PER_HOUR;
+	else if (strategy == BTGreaterStrategyNumber || strategy == BTGreaterEqualStrategyNumber)
+		new_value -= 4 * USECS_PER_HOUR;
+
+	return TimestampTzGetDatum(new_value);
+}
+
 /*
  * Transform time_bucket calls of the following form in WHERE clause:
  *
  * time_bucket(width, column) OP value
+ * time_bucket(width, column, timezone, origin, offset) OP value
```

Comment:
As per docs the correct order of  arguments is (width, column, timezone, offset, origin)
https://docs.timescale.com/api/latest/hyperfunctions/time_bucket/
As per implementation of the function correct order is (width, column, timezone, origin, offset)

I guess docs should be corrected.

Reply:
The documentation is wrong, have a look:
```sql
fabrizio=# SELECT proname, proargnames FROM pg_proc WHERE proname = 'time_bucket' AND array_length(proargnames, 1) = 5;
   proname   |               proargnames                
-------------+------------------------------------------
 time_bucket | {bucket_width,ts,timezone,origin,offset}
(1 row)
```

Reply:
Created a docs PR for it: https://github.com/timescale/docs/pull/2041

Reply:
Thanks.